### PR TITLE
[dart] add theia-dart-docker

### PR DESCRIPTION
**Pull request by satishbabariya**
note: flutter not added. only dart-lang added

### CODE REVIEW

**Comment by satishbabariya** [comment]
for a testing pub run `git clone https://github.com/ReactiveX/rxdart.git`

File: theia-dart-docker/latest.package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+    "private": true,
+    "dependencies": {
```
Comment (vince-fugnitto) [inline theia-dart-docker/latest.package.json]: Can we follow the same idea as previous images and include the following:
- enable the preference `files.enableTrash`
- include an `applicationName` (ex: Theia Dart Example)


_Example:_
https://github.com/theia-ide/theia-apps/blob/27f92587e9cee99616c5c0610aa94c96211504ae/theia-go-docker/latest.package.json#L3-L12

File: theia-dart-docker/latest.package.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        "@theia/debug": "latest"
+    },
+    "scripts": {
+        "postinstall": "download-debug-adapters"
+    },
+    "adapterDir": "plugins",
+    "adapters": {
+        "dart-code": "https://ms-vscode.gallery.vsassets.io/_apis/public/gallery/publisher/Dart-Code/extension/dart-code/3.2.1/assetbyname/Microsoft.VisualStudio.Services.VSIXPackage"
+    }
+}
```
Comment (vince-fugnitto) [inline theia-dart-docker/latest.package.json]: [minor] please keep a new line at the end of files.

File: theia-dart-docker/next.package.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
+    "private": true,
+    "dependencies": {
```
Comment (vince-fugnitto) [inline theia-dart-docker/next.package.json]: Same comment as the `latest.package.json`.

File: theia-dart-docker/latest.package.json (hunk 1 above)
Comment (satishbabariya) [reply theia-dart-docker/latest.package.json]: sure

**Comment by vince-fugnitto** [comment]
@satishbabariya do you mind squashing your PR into a single commit and making sure to `sign-off`?

**Comment by satishbabariya** [comment]
@vince-fugnitto sure

File: theia-dart-docker/latest.package.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
```
Comment (vince-fugnitto) [inline theia-dart-docker/latest.package.json]: [minor] please include a new line at the end of files.

**Comment by vince-fugnitto** [comment]
It looks like there are some conflicts, please rebase the PR against the latest changes, make sure to `sign-off` and squash all commits into a single commit :)

**Comment by satishbabariya** [comment]
@vince-fugnitto done.

**Comment by satishbabariya** [comment]
squashing those two commit will be difficult due to the second commit is a mix of merge old history and new changes.

**Comment by vince-fugnitto** [comment]
> squashing those two commit will be difficult due to the second commit is a mix of merge old history and new changes.

You can revert your changes to master (since it is a single line), and then re-apply your changes.
```
git checkout origin/master .travis.yml
git checkout origin/master tests/package.json
```
Given that your `master` branch is up to date with the latest changes, then you can reapply your changes.

**Comment by vince-fugnitto** [comment]
@satishbabariya would you like me to handle the rebase?

**Comment by satishbabariya** [comment]
@vince-fugnitto sure.

**Comment by vince-fugnitto** [comment]
I've updated the code and rebased it (resolving conflicts).
I'll perform a final sanity check test and merge it for you if it's successful :)

**Comment by vince-fugnitto** [comment]
Thank you for your contribution @satishbabariya 👍

**Pull request merged**